pkg/cmd/auth/status/status_test.go

```diff
@@ -433,6 +440,27 @@ func Test_statusRun(t *testing.T) {
 				  - To forget about this account, run: gh auth logout -h ghe.io -u monalisa-ghe
 			`),
 		},
+		{
+			name: "active",
+			opts: StatusOptions{
+				Active: true,
+			},
+			cfgStubs: func(t *testing.T, c gh.Config) {
+				login(t, c, "github.com", "monalisa", "gho_abc123", "https")
+				login(t, c, "github.com", "monalisa-2", "gho_abc123", "https")
+			},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(httpmock.REST("GET", ""), httpmock.ScopesResponder("repo,read:org"))
+			},
+			wantOut: heredoc.Doc(`
+				github.com
+				  ✓ Logged in to github.com account monalisa-2 (GH_CONFIG_DIR/hosts.yml)
+				  - Active account: true
+				  - Git operations protocol: https
+				  - Token: gho_******
+				  - Token scopes: 'repo', 'read:org'
+			`),
+		},
```

Comment:
**How should this test be expanded to account for inactive user accounts that are invalid?**

If users specify the `--active` flag, we will want to make sure any invalid accounts that are inactive don't result in an err / error exit code.

It should be taking a similar approach as the test about this of stubbing HTTP requests for `monalisa-ghe` except that the output would only include the active account.

Reply:
Hi @andyfeller 

As we only take active accounts into consideration, inactive accounts are automatically skipped, I couldn't able to simulate a situation to validate the token for inactive accounts. 
![image](https://github.com/user-attachments/assets/2e059901-cde9-4ba0-b174-52ac94f48402)

Could you please advice on any approach to simulate the scenario? Thanks.

Reply:
That is a good point 🤔 and I'm not sure there is a way now that you mention it.  `gh` can't tell if a token is valid until it is used, which is why this test simulates an error from the API when it is.  By skipping inactive accounts, there is no way to really test that.

Thank you for considering it and giving it a go ❤️

Reply:
Would the new [`Exclude` method on the registry](https://github.com/cli/cli/blob/192f57ef429b4e89937c87dcbd330520730d4a3a/pkg/httpmock/registry.go#L31-L46) help solve this problem? If you're looking to make sure we don't do anything on inactive accounts, this seems like a reasonable use-case for it.

Apologies if I'm misunderstanding the context, here 🙂

Reply:
> Would the new [Exclude method on the registry](https://github.com/cli/cli/blob/192f57ef429b4e89937c87dcbd330520730d4a3a/pkg/httpmock/registry.go#L31-L46) help solve this problem?

It _could_ work but it's a bit of an overspecification to say we don't want to see this request. The key experience that is under test is that invalid inactive accounts don't result in an exit code of `1`. Whether the API is called or not is a bit of an implementation detail.

The deeper issue here is that the registry is a truly a [mock, when it would be better as a fake]((https://www.martinfowler.com/bliki/TestDouble.html) ). If the server were preconfigured with a mapping of token -> validity, then even if the implementation changd and that API call was made, we would still know the behaviour was upheld.

But I don't think it's worth taking on that work just for this. I think I'd rather see not see Exclude used here either and just leave it alone.